src/components/ProjectSubscription.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useContext, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Context as UserContext } from '../contextProviders/UserProvider';
+import { updateSubscription, inquirySubscriptionStatus } from '../services/subscriptionService';
+import { actionWithLoggedIn } from '../lib/middleware';
+import ErrorHandler from '../lib/ErrorHandler';
+
+function ProjectSubscription({ projectTypeId, projectType }) {
+  const {
+    state: { currentUser },
+  } = useContext(UserContext);
+
+  const [subscribed, setSubscribed] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  const { authenticated } = currentUser;
+
+  const toggleSubscription = async enabled => {
+    if (!authenticated) {
+      React.signIn();
+      return;
+    }
+    setLoading(true);
+    await actionWithLoggedIn(currentUser);
```

Comment:
It's not necessary, upper condition on authenticated is doing the same job

Reply:
done